File: maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum = nums[0]
        maxSum = nums[0]
        for i in range(1, len(nums)):
            sum = max(nums[i], sum + nums[i])
            maxSum = max(maxSum, sum)
        return maxSum

File: maximum-subarray/maximum-subarray.c
#define MAX(a,b) (((a)>(b))?(a):(b))

int maxSubArray(int* nums, int numsSize){
    int sum = nums[0];
    int maxSum = nums[0];
    for(int i = 1; i < numsSize; i++) {
        sum = MAX(nums[i], sum + nums[i]);
        maxSum = MAX(maxSum, sum);
    }
    return maxSum;
